dashboard/app/controllers/api/v1/pd/workshop_enrollments_controller.rb

```diff
@@ -102,6 +102,16 @@ def cancel
     Pd::WorkshopMailer.organizer_cancel_receipt(enrollment).deliver_now
   end
 
+  # POST /api/v1/pd/workshops/1/enrollments/move
+  def move
+    params[:enrollment_ids].each do |id|
+      enrollment = @workshop.enrollments.find_by(id: id)
+      enrollment.attendances.destroy_all
+      enrollment.update(pd_workshop_id: params[:destination_workshop_id])
+    end
```

Comment:
You can do much of this without a loop, reducing the number of queries (maybe premature optimization though).  Would also be good to do the whole thing in a transaction, so you don't end up destroying attendances and then not moving the enrollment.

```rb
Pd::Enrollment.transaction do
  # Find the enrollments in one query
  enrollments = Pd::Enrollment.where(id: params[:enrollment_ids])

  # Delete attendances in one query
  # no need for validations and callbacks when soft-deleting attendance records
  Pd::Attendance.where(pd_enrollment: enrollments).delete_all

  # We do want to make sure callbacks run on the enrollment updates,
  # so use many `update` calls instead of `update_all`
  enrollments.each {|e| e.update(pd_workshop_id: params[:destination_workshop_id])}
end
```